migrate.py

```diff
@@ -649,6 +690,314 @@ def copy_tests(plugin, coll, spec, args):
                         with open(yf, 'w') as f:
                             f.write(ydata)
 
+##############################################################################
+# Rewrite integration tests
+##############################################################################
+
+integration_tests_deps = set()
+
+def integration_tests_add_to_deps(collection, dep_collection):
+    global integration_tests_deps
+    integration_tests_deps.add(dep_collection)
+    logger.debug("Adding " + dep_collection + " as a dep for " + collection)
+
+
+def poor_mans_integration_tests_discovery(checkout_dir, plugin_type, plugin_name):
+    # FIXME this might be actually enough for modules integration tests, at least for the most part
+    if plugin_type != 'modules':
+        return []
+
+    files = glob.glob(os.path.join(checkout_dir, 'test/integration/targets', os.path.basename(os.path.splitext(plugin_name)[0])))
+    for fname in files:
+        logger.debug('Found integration tests for %s %s in %s' % (plugin_type, plugin_name, fname))
+
+    return files
+
+
+def rewrite_integration_tests(test_dirs, checkout_dir, collection_dir, namespace, collection, spec, args):
+    # FIXME move to diff file
+    # FIXME module_defaults groups
+
+    for test_dir in test_dirs:
```

Comment:
Let's reduce a few levels of nesting:
```suggestion
    filenames = (
        os.path.join(dp, f)
        for td in test_dirs
        for (dp, _dn, fn) in os.walk(td)
        for f in fn
    )
    for full_path in filenames:
```